### PR TITLE
Support for new ad placement locations on home page feed

### PR DESCRIPTION
**Pull request by rt4914**
## What type of PR is this? (check all applicable)

- [x] Feature

## Description

Tasks finished from issues:

- [x] feed_first - Before all home page posts
- [x] feed_second - Between 2nd and 3rd posts in the feed
- [x] feed_third - Between 7th and 8th posts in the feed
- [x] these options should be available at /admin/customization/display_ads/new
- [ ] these options should appear in API documentation for display_ads **( Implemented in #19306)**
- [x] remove hard-coded sign_in_invitation (aka CTA)
- [x] remove [event](https://github.com/forem/forem/blob/09291b420515d265eca031b431a5937bc63c3bb3/app/javascript/packs/homePage.jsx#L135) as we won't be tracking clicks on this CTA.

## Related Tickets & Documents

- Related Issue #19225

## QA Instructions, Screenshots, Recordings

[Figma](https://www.figma.com/file/9UfSjNYJs745w3Q8JCUqsL/Small-Wins---%E2%80%A8Core-Experience-%E2%9A%A1%EF%B8%8F?node-id=1623-7063&t=fvJIv0Rh0lEk374A-0)

Loom Recording: https://www.loom.com/share/92f295adf4c34a89a20efb30e4cdf878
There was one minor bug mentioned in above video it has been fixed: https://www.loom.com/share/a5a4cf28843e4215b86833b133ca9a93

**How to test?**

1. Checkout in this branch and sign in as admin locally
2. Create three new billboards for position "Home Feed First", "Home Feed Second" and "Home Feed Third"
3. And either sign-out or open home page in incognito mode
4. Check the three billboards that you created at various positions.

**Output**

Screenshot 2023-04-07 at 3 01 34 PM
Screenshot 2023-04-07 at 3 01 49 PM
Screenshot 2023-04-07 at 3 01 59 PM

### UI accessibility concerns?

No

## Added/updated tests?

- [x] No, need help with exactly where can I add these tests and also add display ads to the feed.

### CODE REVIEW

File: app/views/articles/show.html.erb
Comment (benhalpern) [inline app/views/articles/show.html.erb:204]: Since we already pass `@display_ad` into the partial, I feel like we could access the context of whether this is "displayed_in_home_feed" via that instead of having to add it to all the partials, what do you think?

Comment (rt4914) [reply app/views/articles/show.html.erb]: Thanks @benhalpern I have removed `displayed_in_home_feed ` and instead used `display_ad.placement_area`.

Let me know if it can be further improved.

**Review by jaw6 — APPROVED** [review]
Tested locally, looks good to me :+1:

**Review by benhalpern — APPROVED** [review]
LGTM, tested on Uffizzi 😋

File: app/views/shared/_display_ad.html.erb
Comment (benhalpern) [inline app/views/shared/_display_ad.html.erb:1]: As a non-blocking suggestion, this could maybe be written out as `display_ad.placement_area.start_with?("feed_")` which might be just a bit cleaner.

Comment (rt4914) [reply app/views/shared/_display_ad.html.erb]: Applied this change. Thanks.

**Comment by rt4914** [comment]
Will wait for one more review from either one @Ridhwana / @lboogie04  / @lightalloy as this PR introduces some key changes to home feed.

**Comment by Ridhwana** [comment]
> Will wait for one more review from either one @Ridhwana / @lboogie04 / @lightalloy as this PR introduces some key changes to home feed.

@rt4914 I've been offline and have a bit of catchup to do but if nobody else reviews this by tomorrow then I'll have a look.

**Comment by lightalloy** [comment]
I won't be able to review today 🌝

**Review by Ridhwana — APPROVED** [review]
LGTM 🚀

**Pull request merged**
